src/dex/cables/cables.ts

```diff
@@ -0,0 +1,1006 @@
+      // might be overwritten on Executors
+      quoteData.takerAmount,
+    ];
+
+    const exchangeData = this.rfqInterface.encodeFunctionData(
+      swapFunction,
+      swapFunctionParams,
+    );
+
+    const fromAmount = ethers.utils.defaultAbiCoder.encode(
+      ['uint256'],
+      [quoteData.takerAmount],
+    );
+
+    const filledAmountIndex = exchangeData
+      .replace('0x', '')
+      .lastIndexOf(fromAmount.replace('0x', ''));
+
+    const filledAmountPos =
+      (filledAmountIndex !== -1 ? filledAmountIndex : exchangeData.length) / 2;
+
+    return {
+      exchangeData,
+      needWrapNative: this.needWrapNative,
+      dexFuncHasRecipient: false,
+      targetExchange: this.mainnetRFQAddress,
+      returnAmountPos: undefined,
+      insertFromAmountPos: filledAmountPos,
+    };
+  }
+
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: CablesData,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    const { quoteData } = data;
+
+    assert(
+      quoteData !== undefined,
+      `${this.dexKey}-${this.network}: quoteData undefined`,
+    );
+
+    const params = [
+      {
+        nonceAndMeta: quoteData.nonceAndMeta,
+        expiry: quoteData.expiry,
+        makerAsset: quoteData.makerAsset,
+        takerAsset: quoteData.takerAsset,
+        maker: quoteData.maker,
+        taker: quoteData.taker,
+        makerAmount: quoteData.makerAmount,
+        takerAmount: quoteData.takerAmount,
+      },
+      quoteData.signature,
+    ];
+
+    const payload = this.abiCoder.encodeParameter(
+      {
+        ParentStruct: {
+          order: {
+            nonceAndMeta: 'uint256',
+            expiry: 'uint128',
+            makerAsset: 'address',
+            takerAsset: 'address',
+            maker: 'address',
+            taker: 'address',
+            makerAmount: 'uint256',
+            takerAmount: 'uint256',
+          },
+          signature: 'bytes',
+        },
+      },
+      {
+        order: params[0],
+        signature: params[1],
+      },
+    );
+
+    return {
+      targetExchange: this.mainnetRFQAddress,
+      payload,
+      networkFee: '0',
+    };
+  }
+
+  normalizeToken(token: Token): Token {
+    return {
+      ...token,
+      address: this.normalizeTokenAddress(token.address),
+    };
+  }
+
+  normalizeTokenAddress(address: Address): Address {
+    return address.toLowerCase();
+  }
+
+  getTokenFromAddress(address: Address): Token {
+    return this.tokensMap[this.normalizeAddress(address)];
+  }
+
+  getPoolIdentifier(srcAddress: Address, destAddress: Address, mm?: string) {
+    return `${this.dexKey}_${srcAddress}_${destAddress}`.toLowerCase();
+    return `${this.dexKey}_${srcAddress}_${destAddress}_${mm}`.toLowerCase();
+  }
+
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    await this.setTokensMap();
```

Comment:
let's do Redis cache call after if checks that we do bellow (for empty tokens and same tokens)

Reply:
fixed